### PR TITLE
[2.0] Drop Python 2.7, 3.5 from CI (1)

### PR DESCRIPTION
**Pull request by sentrivana**
- remove special treatment for Python 2.7 from CI YAML generation
- remove Python 2.7, 3.5 test environments from `tox.ini`
- remove tests for framework versions that don't support at least Python 3.6

---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

File: .github/workflows/test-integrations-miscellaneous.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.5","3.8","3.11","3.12"]
+        python-version: ["3.8","3.11","3.12"]
```
Comment (szokeasaurusrex) [inline .github/workflows/test-integrations-miscellaneous.yml:28]: Should we be testing 3.6 and/or 3.7 here?

File: tox.ini [hunk 2]
```diff
@@ -52,18 +52,17 @@ envlist =
...
-    {py2.7,py3.6,py3.7}-boto3-v{1.12}
+    {py3.6,py3.7}-boto3-v{1.12}
     {py3.7,py3.11,py3.12}-boto3-v{1.21}
     {py3.7,py3.11,py3.12}-boto3-v{1.29}
     {py3.7,py3.11,py3.12}-boto3-latest
 
     # Bottle
-    {py2.7,py3.5,py3.9}-bottle-v{0.12}
-    {py3.5,py3.11,py3.12}-bottle-latest
+    {py3.9}-bottle-v{0.12}
```
Comment (szokeasaurusrex) [inline tox.ini:61]: Do we need py 3.6 here (or in any of the other tests where py3.5 was removed)?

**Review by szokeasaurusrex — COMMENTED** [review]
I have a couple questions, please see inline comments

File: tox.ini (hunk 2 above)
Comment (sentrivana) [reply tox.ini]: This is a very good point, adding this!

File: .github/workflows/test-integrations-miscellaneous.yml (hunk 1 above)
Comment (sentrivana) [reply .github/workflows/test-integrations-miscellaneous.yml]: We should -- the yaml files will be updated as soon as I add these to tox.ini

**Comment by sentrivana** [comment]
Added `3.6` as the lowest supported version to test against in places where we're removing 3.5 -- please re-review @szokeasaurusrex 🙏🏻

File: tox.ini [hunk 3]
```diff
@@ -139,29 +136,29 @@ envlist =
...
-    {py3.5,py3.11,py3.12}-loguru-latest
+    {py3.6,py3.11,py3.12}-loguru-v{0.5}
+    {py3.6,py3.11,py3.12}-loguru-latest
 
     # OpenTelemetry (OTel)
     {py3.7,py3.9,py3.11,py3.12}-opentelemetry
 
     # pure_eval
-    {py3.5,py3.11,py3.12}-pure_eval
+    {py3.11,py3.12}-pure_eval
```
Comment (szokeasaurusrex) [inline tox.ini:150]: 3.6?

File: tox.ini [hunk 4]
```diff
@@ -171,27 +168,27 @@ envlist =
     {py3.8,py3.11,py3.12}-quart-latest
 
     # Redis
-    {py2.7,py3.7,py3.8}-redis-v{3}
+    {py3.7,py3.8}-redis-v{3}
```
Comment (szokeasaurusrex) [inline tox.ini:171]: 3.6?

File: tox.ini [hunk 5]
```diff
@@ -171,27 +168,27 @@ envlist =
...
     {py3.7,py3.8,py3.11}-redis-v{4}
     {py3.7,py3.11,py3.12}-redis-v{5}
     {py3.7,py3.11,py3.12}-redis-latest
 
     # Redis Cluster
-    {py2.7,py3.7,py3.8}-rediscluster-v{1,2}
+    {py3.7,py3.8}-rediscluster-v{1,2}
```
Comment (szokeasaurusrex) [inline tox.ini:177]: 3.6?

**Review by szokeasaurusrex — APPROVED** [review]
I added inline comments in a few spots where it is possible that we should be adding 3.6, but I am unsure. Please double check these locations, and fix them if necessary – then I think this is good to merge!

File: tox.ini (hunk 3 above)
Comment (sentrivana) [reply tox.ini:150]: Missed this one, will add

Comment (sentrivana) [reply tox.ini]: Changed to `{py3.6,py3.11,py3.12}-pure_eval`

File: tox.ini (hunk 5 above)
Comment (sentrivana) [reply tox.ini]: Changed to `{py3.6,py3.8}-rediscluster-v{1,2}` (same reasoning as above)

File: tox.ini [hunk 6]
```diff
@@ -171,27 +168,27 @@ envlist =
...
     # no -latest, not developed anymore
 
     # Requests
-    {py2.7,py3.8,py3.11,py3.12}-requests
+    {py3.8,py3.11,py3.12}-requests
```
[the remark this answers is not in the document]
Comment (sentrivana) [reply tox.ini]: 👍🏻 `{py3.6,py3.8,py3.11,py3.12}-requests`

File: tox.ini [hunk 7]
```diff
@@ -208,7 +205,7 @@ envlist =
     # 1.51.14 is the last starlite version; the project continues as litestar
 
     # SQL Alchemy
-    {py2.7,py3.7,py3.9}-sqlalchemy-v{1.2,1.4}
+    {py3.7,py3.9}-sqlalchemy-v{1.2,1.4}
```
[the remark this answers is not in the document]
Comment (sentrivana) [reply tox.ini]: Changed to `{py3.6,py3.9}-sqlalchemy-v{1.2,1.4}`

**Comment by sentrivana** [comment]
Thanks for the thorough review @szokeasaurusrex! 👍🏻

**Pull request merged**